### PR TITLE
Middleware for protecting metrics endpoint

### PR DESCRIPTION
**Pull request by revol-xut**
## Required Doc/Changelog Updates

- [ ] **I HAVE UPDATED CHANGELOG ACCORDINGLY**
- [ ] **I HAVE UPDATED DOCS ACCORDINGLY**

### CODE REVIEW

File: flake.lock
Comment (oxapentane) [inline flake.lock:1]: Split out lock bumps to separate commit

File: flake.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
```
Comment (oxapentane) [inline flake.nix:3]: Why? Our deployments are targeting stable. Also, not part of "implementing middleware"

File: src/security.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+{
+    type Response = ServiceResponse<EitherBody<B>>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = ExportAuthentificationMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        let raw_address_ranges: Vec<String> = serde_json::from_str(
+            &std::env::var("PROMETHEUS_ALLOWED_IPS").unwrap_or("[\"10.13.37.0/24\"]".to_string()),
```
Comment (oxapentane) [inline src/security.rs:30]: No defaults. Not configured -> all requests to the endpoint gracefully HTTP error out

Comment (oxapentane) [inline src/security.rs:30]: DATACARE_PROMETHEUS_ALLOWED_IPS

File: src/security.rs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        )
+        .unwrap_or(vec!["10.13.37.0/24".to_string()]);
+        let allowed_ranges: Vec<IpNet> = raw_address_ranges
+            .iter()
+            .map(|x| x.parse().unwrap())
```
Comment (oxapentane) [inline src/security.rs:35]:
```suggestion
            .map(|x| x.parse().expect("meaningful error message"))
```

File: src/security.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    type Error = Error;
+    type InitError = ();
+    type Transform = ExportAuthentificationMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        let raw_address_ranges: Vec<String> = serde_json::from_str(
+            &std::env::var("PROMETHEUS_ALLOWED_IPS").unwrap_or("[\"10.13.37.0/24\"]".to_string()),
+        )
+        .unwrap_or(vec!["10.13.37.0/24".to_string()]);
```
Comment (oxapentane) [inline src/security.rs:32]: No defaults. Match, throw a warning, return HTTP error on any request to the endpoint

**Review by oxapentane — CHANGES_REQUESTED** [review]
See comments

File: flake.nix (hunk 1 above)
Comment (revol-xut) [reply flake.nix:3]: some of the crates need a very version of rust and this was the easiest to get it to build again.

File: src/security.rs (hunk 2 above)
Comment (revol-xut) [reply src/security.rs:30]: So u just want an expect(...) ?

Comment (revol-xut) [reply src/security.rs:30]: fair

Comment (oxapentane) [reply src/security.rs:30]: also possible, like if we want to just configure nothing or hard fail (expect) on missing config - up to you

**Review by oxapentane — CHANGES_REQUESTED** [review]
fix the history

**Review by oxapentane — APPROVED** [review]

**Pull request merged**